### PR TITLE
chore: add reproduction for issue #21352

### LINKED ISSUE #21352: Invalid `….findMany()` invocation: The column `j1.id` does not exist in the current database.
### Bug description

After updating to 5.4.0, the following error is shown when running the command provided below with the schema provided below:

```
prisma:query SELECT "public"."Post"."id" FROM "public"."Post" LEFT JOIN "public"."User" AS "j1" ON ("j1"."email") = ("public"."Post"."email") WHERE ("j1"."email" = $1 AND ("j1"."id" IS NOT NULL)) OFFSET $2
prisma:error 
Invalid `prisma.post.findMany()` invocation in
…/prisma21352/src/index.ts:14:33
…
→ 14 console.log(await prisma.post.findMany(
The column `j1.id` does not exist in the current database.
```

### How to reproduce

**See also: https://github.com/marvinruder/prisma21352**

Run
```js
await prisma.post.findMany({
  select: {
    id: true,
  },
  where: {
    user: {
      email: 'alice@example.com',
    }
  }
})
```
with client generated from the provided schema. 

### Expected behavior

All `watchlist` rows related to the specified user (identified by their email) are returned.

### Prisma information

```prisma
// Add your schema.prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model User {
  email String      @id @db.VarChar(255)
  posts Post[]
}

model Post {
  id    Int     @id @default(autoincrement())
  user  User    @relation(fields: [email], references: [email], onUpdate: Cascade, onDelete: Cascade)
  email String  @db.VarChar(255)
}
```

```ts
// Add your code using Prisma Client
await prisma.post.findMany({
  select: {
    id: true,
  },
  where: {
    user: {
      email: 'alice@example.com',
    }
  }
})
```

### Environment & setup

- OS: occurring on macOS 14.0 (build 23A344) and in `node:20.8.0-alpine` Docker image
- Database: PostgreSQL
- Node.js version: 20.8.0

### Prisma Version

```
prisma                  : 5.4.0
@prisma/client          : 5.4.0
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine a5596b96668f0f4b397761ce0956db54e17e48c4 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Schema Engine           : schema-engine-cli a5596b96668f0f4b397761ce0956db54e17e48c4 (at node_modules/@prisma/engines/schema-engine-darwin-arm64)
Schema Wasm             : @prisma/prisma-schema-wasm 5.4.0-47.a5596b96668f0f4b397761ce0956db54e17e48c4
Default Engines Hash    : a5596b96668f0f4b397761ce0956db54e17e48c4
Studio                  : 0.494.0
```

### PR DESCRIPTION
**Pull request by jkomyno**
This PR was tested locally with https://github.com/prisma/prisma-engines/pull/4315, confirming that https://github.com/prisma/prisma-engines/pull/4315 indeed fixes #21352.

### CODE REVIEW

File: packages/client/tests/functional/issues/21352-id-does-not-exist/tests.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(
+  () => {
+    // TODO: this currently fails with "The column `j1.id` does not exist in the current database.".
+    // Close https://github.com/prisma/prisma/issues/21352 once fixed.
+    test('[1] should not fail', async () => {
```
Comment (millsp) [inline packages/client/tests/functional/issues/21352-id-does-not-exist/tests.ts:11]:
```suggestion
    test.failing('[1] should not fail', async () => {
```

File: packages/client/tests/functional/issues/21352-id-does-not-exist/tests.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+          user: {
+            email: 'info@example.com',
+          },
+        },
+      })
+    })
+
+    // TODO: this currently fails with "The column `j1.field` does not exist in the current database.".
+    // Close https://github.com/prisma/prisma/issues/21352 once fixed.
+    test('[2] should not fail', async () => {
```
Comment (millsp) [inline packages/client/tests/functional/issues/21352-id-does-not-exist/tests.ts:26]:
```suggestion
    test.failing('[2] should not fail', async () => {
```

Comment (millsp) [reply packages/client/tests/functional/issues/21352-id-does-not-exist/tests.ts:26]: We can also wait for the engines to be merged

Comment (jkomyno) [reply packages/client/tests/functional/issues/21352-id-does-not-exist/tests.ts:26]: > We can also wait for the engines to be merged
That'd be the plan, yeah.

**Review by millsp — APPROVED** [review]

**Pull request merged**